<a href="https://colab.research.google.com/github/Souptikbej/Ecommerce-Sales-EDA/blob/main/notebooks/Ecommerce_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **E-Commerce Sales Analysis**

## Phase 2: Dataset Understanding

### Objective

The objective of this phase is to understand the dataset before performing any data cleaning or analysis.

During this phase, we will:

- Load the dataset
- Inspect its structure
- Understand each column
- Identify data types
- Explore summary statistics
- Build a data dictionary
- Detect potential data quality issues

Understanding the dataset helps ensure accurate and meaningful analysis in later stages.

In [1]:
import pandas as pd
import numpy as np

## 1. Load Dataset

In [3]:
df = pd.read_csv("SampleSuperstore_Practice.csv")

df.head()

,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,CA-10001,2023-11-24,2023-11-30,Standard Class,CUST-704,Customer 55,Consumer,USA,Los Angeles,California,West,P-1488,Furniture,Bookcases,Bookcases Model 3,564.56,2,0.1,-26.65
1,CA-10002,2023-03-06,2023-03-08,Same Day,CUST-703,Customer 36,Consumer,USA,New York,New York,East,P-3615,Furniture,Bookcases,Bookcases Model 14,1212.00,9,0.1,257.20
2,CA-10003,2022-04-10,2022-04-13,First Class,CUST-718,Customer 34,Consumer,USA,Houston,Texas,South,P-8527,Office Supplies,Binders,Binders Model 18,446.32,7,0.2,-33.03
3,CA-10004,2022-02-16,2022-02-22,Second Class,CUST-891,Customer 38,Consumer,USA,Los Angeles,California,West,P-4814,Office Supplies,Binders,Binders Model 4,1113.04,2,0.2,155.61
4,CA-10005,2022-08-03,2022-08-09,First Class,CUST-818,Customer 10,Home Office,USA,Chicago,Illinois,Central,P-3803,Office Supplies,Paper,Paper Model 18,1278.66,6,0.2,-59.85


## 2. Dataset Shape

Understanding the number of rows and columns helps estimate the size of the dataset.

In [5]:
row=df.shape[0]
column=df.shape[1]
print(f"The dataset has {row} rows and {column} columns")

The dataset has 200 rows and 19 columns


## 3. Column Names

Listing all columns helps understand what information is available.

In [6]:
df.columns

Index(['Order ID', 'Order Date', 'Ship Date', 'Ship Mode', 'Customer ID',
       'Customer Name', 'Segment', 'Country', 'City', 'State', 'Region',
       'Product ID', 'Category', 'Sub-Category', 'Product Name', 'Sales',
       'Quantity', 'Discount', 'Profit'],
      dtype='object')

### Observation

The dataset contains information related to:

- Orders
- Customers
- Products
- Sales
- Shipping
- Location
- Profit

## 4. Preview Dataset

Viewing the first few records helps understand how the data is organized.

In [7]:
df.head()

,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,CA-10001,2023-11-24,2023-11-30,Standard Class,CUST-704,Customer 55,Consumer,USA,Los Angeles,California,West,P-1488,Furniture,Bookcases,Bookcases Model 3,564.56,2,0.1,-26.65
1,CA-10002,2023-03-06,2023-03-08,Same Day,CUST-703,Customer 36,Consumer,USA,New York,New York,East,P-3615,Furniture,Bookcases,Bookcases Model 14,1212.00,9,0.1,257.20
2,CA-10003,2022-04-10,2022-04-13,First Class,CUST-718,Customer 34,Consumer,USA,Houston,Texas,South,P-8527,Office Supplies,Binders,Binders Model 18,446.32,7,0.2,-33.03
3,CA-10004,2022-02-16,2022-02-22,Second Class,CUST-891,Customer 38,Consumer,USA,Los Angeles,California,West,P-4814,Office Supplies,Binders,Binders Model 4,1113.04,2,0.2,155.61
4,CA-10005,2022-08-03,2022-08-09,First Class,CUST-818,Customer 10,Home Office,USA,Chicago,Illinois,Central,P-3803,Office Supplies,Paper,Paper Model 18,1278.66,6,0.2,-59.85


### Observation

- Each row represents one sales transaction.

- Each column describes a specific attribute of that transaction.

## 5. Dataset Information

Checking data types and missing values.

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 19 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Order ID       200 non-null    object 
 1   Order Date     200 non-null    object 
 2   Ship Date      200 non-null    object 
 3   Ship Mode      200 non-null    object 
 4   Customer ID    200 non-null    object 
 5   Customer Name  200 non-null    object 
 6   Segment        200 non-null    object 
 7   Country        200 non-null    object 
 8   City           200 non-null    object 
 9   State          200 non-null    object 
 10  Region         200 non-null    object 
 11  Product ID     200 non-null    object 
 12  Category       200 non-null    object 
 13  Sub-Category   200 non-null    object 
 14  Product Name   200 non-null    object 
 15  Sales          200 non-null    float64
 16  Quantity       200 non-null    int64  
 17  Discount       200 non-null    float64
 18  Profit    

### Observation

- Numerical Columns: 4
  - Sales (float64)
  - Quantity (int64)
  - Discount (float64)
  - Profit (float64)

- Categorical Columns: 13
  - Order ID
  - Ship Mode
  - Customer ID
  - Customer Name
  - Segment
  - Country
  - City
  - State
  - Region
  - Product ID
  - Category
  - Sub-Category
  - Product Name

- Date Columns: 2 (currently stored as object)
  - Order Date
  - Ship Date

- Missing Values: No missing values
  - All 19 columns have 200 non-null values.

## 6. Summary Statistics

Understanding the distribution of numerical variables.

In [9]:
df.describe()

,Sales,Quantity,Discount,Profit
count,200.000000,200.000000,200.000000,200.000000
mean,1020.232900,5.710000,0.158500,114.393050
std,551.840916,2.782916,0.109946,149.357508
min,38.440000,1.000000,0.000000,-197.940000
25%,563.680000,3.000000,0.100000,11.740000
50%,1008.225000,6.000000,0.200000,69.410000
75%,1486.025000,8.000000,0.300000,206.872500
max,1996.280000,10.000000,0.300000,597.310000


### Observation

- **Average Sales:** **1020.23**
  - On average, each order generated sales of approximately **1020.23** units.

- **Maximum Sales:** **1996.28**
  - The highest sales value recorded in the dataset is **1996.28**, while the minimum is **38.44**.

- **Profit Distribution:**
  - Average Profit: **114.39**
  - Median Profit: **69.41**
  - Profit ranges from **-197.94** (loss) to **597.31** (highest profit).
  - The presence of negative profit values indicates that some orders resulted in a loss.

- **Discount Range:**
  - Discounts vary from **0%** to **30%**.
  - The average discount is **15.85%**, with **20%** being the most common (median) discount.

- **Quantity Distribution:**
  - Customers purchased between **1 and 10** units per order.
  - The average quantity ordered is **5.71** units.
  - Half of the orders contain **6 or fewer** items (median = 6).

## 7. Data Dictionary

| Column      | Data Type | Description                |
| ----------- | --------- | -------------------------- |
| Order ID    | Object    | Unique order identifier    |
| Customer ID | Object    | Unique customer identifier |
| Sales       | Float     | Revenue generated          |
| Profit      | Float     | Profit earned              |
| Quantity    | Integer   | Number of units sold       |


## 8. Initial Findings

### Key Findings

- Dataset contains 200 records.
- Dataset contains 19 columns.
- Date columns may require conversion.
- Sales and Profit are numerical variables.
- Customer and Product information are categorical variables.

### Potential Data Quality Issues

- Missing values: To be checked in the next phase.
- Duplicate records: To be checked in the next phase.
- Incorrect data types: To be validated during data cleaning.